h2/src/main/org/h2/engine/Database.java

```diff
@@ -1270,9 +1270,11 @@ private synchronized void closeAllSessionsExcept(SessionLocal except) {
 
         int timeout = 2 * getLockTimeout();
         long start = System.currentTimeMillis();
+        // 'sleep' should be strictly greater than zero, otherwise real time is not taken into consideration
+        // and the thread simply waits until notified
+        long sleep = Math.max(timeout / 20, 50);
```

Comment:
Thank you for your contribution!

Just replace 0 with 1 here, 50 can be unreasonably large for systems that support small wait values properly.

Reply:
OK.